geode-core/src/integrationTest/java/org/apache/geode/cache/query/internal/index/PartitionedIndexJUnitTest.java

```diff
@@ -66,4 +62,37 @@ public void mapIndexKeysMustContainTheCorrectNumberOfKeysWhenThereIsConcurrentAc
     assertEquals(DATA_SIZE_TO_BE_POPULATED, partitionedIndex.mapIndexKeys.size());
 
   }
+
+  @Test
+  public void verifyAddToAndRemoveFromBucketIndexesUpdatesArbitraryBucketIndex() {
```

Comment:
Would it be possible to also add a test case (or expand this one slightly) that shows that if there are multiple indexes and the current `arbitraryBucketIndex` is removed, that the value of `arbitraryBucketIndex` is reset to another existing index rather than null?